make_a_plea/middleware.py

```diff
@@ -34,8 +39,22 @@ def process_response(self, request, response):
 
 
 class AdminLocaleURLMiddleware(object):
+
     def process_request(self, request):
         if request.path.startswith('/admin'):
             request.LANG = getattr(settings, 'ADMIN_LANGUAGE_CODE', settings.LANGUAGE_CODE)
             translation.activate(request.LANG)
             request.LANGUAGE_CODE = request.LANG
+
+
+class BadRequestExceptionMiddleware(object):
+
+    def process_exception(self, request, e):
+        if isinstance(e, BadRequestException):
+            return HttpResponse(
+                json.dumps(
+                    {"error": e.message},
+                    indent=4),
+                status=400,
+                content_type="application/json",
+            )
```

Comment:
It's not JSON specific. It's for any bad request we want to handle consistently. One could imagine the middleware doing the content type detection based on the incoming accept headers. So it's not about the response being JSON, it's about using exception middleware.